pimd/pim_pim.c

```diff
@@ -585,14 +585,67 @@ static int pim_msg_send_frame(int fd, char *buf, size_t len,
 		return pim_msg_send_frame(fd, (char *)ip2, sendlen, dst, salen,
 					  ifname);
 	}
-#endif
 
 	zlog_warn(
 		"%s: sendto() failure to %pSU: iface=%s fd=%d msg_size=%zd: %m",
 		__func__, dst, ifname, fd, len);
 	return -1;
 }
 
+#else
+static int pim_msg_send_frame(pim_addr src, pim_addr dst, ifindex_t ifindex,
+			      struct iovec *message, int fd)
+{
+	int retval;
+	struct msghdr smsghdr;
+	struct cmsghdr *scmsgp;
+	union {
+		struct cmsghdr hdr;
+		uint8_t buf[CMSG_SPACE(sizeof(struct in6_pktinfo))];
+	} cmsgbuf;
+	struct in6_pktinfo *pktinfo;
+	struct sockaddr_in6 dst_sin6;
+
+	memset(&cmsgbuf, 0, sizeof(cmsgbuf));
```

Comment:
Also instead of the memsets here, just use ` = {}` in the variable definition above.

Reply:
Have made the changes

Reply:
Have made the changes